### ISSUE TITLE
vNext: [IL2CPP] deployment failure (DEP0700) "splash screen image cannot be located"

### ISSUE THREAD

**Issue by davidkline-ms** [Microsoft/MixedRealityToolkit-Unity#2631]
## Overview
Error		DEP0700: Registration of the app failed. [0x80073CF6] AppxManifest.xml(28,27): error 0x80070002: Cannot install or update package Microsoft.MixedReality.Toolkit_ph1m9x8skttmg because the splash screen image [SplashScreen.png] cannot be located. Verify that the package contains an image that can be used as a splash screen for the application, and that the package manifest points to the correct location in the package where this splash screen image can be found.

## Expected Behavior
Can deploy

## Actual Behavior
Cannot deploy

## Steps to reproduce
- Clone repository
- Open project
- Pick favorite scene
- Build for UWP
- Load sln in VS
- Build/deploy (HoloLens or Immersive Headset)

## Unity Editor Version
2018.1.9f1

## Mixed Reality Toolkit Release Version
mrtk_development

**Comment by StephenHodgson** [comment]
Sounds like we didn't set a splash screen in the project settings.

**Comment by davidkline-ms** [comment]
The project looks ok. the asset is there and the manifest looks right... It builds and deploys fine using the .net backend

**Comment by StephenHodgson** [comment]
Wondering if this might actually be a unity bug. Seems like the asset is copied into the directory properly but not referenced by the generated solution.

**Comment by StephenHodgson** [comment]
Yes, confirmed, I'm getting this error in an empty project with IL2CPP.

**Comment by Naphier** [comment]
Is there any workaround for this? I'm not actually using the MR toolkit, but just exported an IL2CPP VS project from Unity and got this error. We'd like to use IL2CPP with Vuforia to have (hopefully) faster tracking speeds.

**Comment by StubbornMinion** [comment]
Having the same issue, but no luck with the above workaround. I've reset the splash screen image in the manifest (although it seemed to be set correctly already), and checked the files are in the Assets folder. I even tried copying the Package.appxmanifest file a .net build folder to the IL2CPP build, but no luck. I am using the MRTK and Vuforia, though I realise this is considered a Unity issue. Can anyone please advise?

Attached is a screengrab of the splash screen settings in VS.

![image](https://user-images.githubusercontent.com/17459370/45256284-653ba080-b38c-11e8-845f-ab0a6766d063.png)

**Comment by Ecnassianer** [comment]
Here's the Unity bug tracking the splashscreen issue:
https://issuetracker.unity3d.com/issues/uwp-cpp-projects-report-not-having-a-splash-screen-set-which-fails-deployment

There are plans to backport to 2017.4 and 2018.2:
https://fogbugz.unity3d.com/default.asp?1082836_hvro1nvjegdd9bjt

**Comment by StephenHodgson** [comment]
Here's a work around:
```
        [PostProcessBuild(2)]
        public static void SplashScreenFix(BuildTarget target, string pathToBuild)
        {
            if (target != BuildTarget.WSAPlayer ||
                PlayerSettings.GetScriptingBackend(BuildTargetGroup.WSA) != ScriptingImplementation.IL2CPP)
            {
                return;
            }

            var vcxProj = Directory.GetFiles(pathToBuild, "*.vcxproj", SearchOption.AllDirectories)
                .Where(project => project.IndexOf("il2cppoutputproject", StringComparison.CurrentCultureIgnoreCase) < 0);
            var projects = vcxProj as string[];

            Assert.IsTrue(projects != null);
            Assert.IsTrue(projects.Length == 1);

            var vcxProject = projects[0];
            var projXml = XDocument.Load(vcxProject);
            var pngElements = projXml.Descendants().Where(x =>
            {
                var val = x.FirstAttribute?.Value.Contains(".png");
                return val.HasValue && val.Value;
            });

            foreach (var element in pngElements)
            {
                foreach (var child in element.Elements())
                {
                    if (child.Name.ToString().Contains("ExcludeFromResourceIndex"))
                    {
                        child.Remove();
                    }
                }
            }

            projXml.Save(vcxProject);
        }
```

**Comment by XPav** [comment]
Thanks much, where should I put this?

**Comment by StephenHodgson** [comment]
Any editor script

**Comment by CasaSpider** [comment]
> I'm having this problem on the non-vNext MRTK also, I _think_ its a Unity bug. A way to solve it is:
> 
> 1. Copy UWP/Proj/Assets/SplashScreen.Scale-200.png to UWP/Proj/Assets/SplashScreen.png
> 2. Add this "new" SplashScreen.png to the project.

Worked for me, thanks!

**Comment by rob100** [comment]
In the Unity 2018.2.9 release should be the fix:

> (1073029 (1071331)) - Universal Windows Platform: Fixed "SplashScreen is missing" error when deploying UWP apps with Visual Studio 15.8.

I will update and give it a try.

**Comment by timkeo** [comment]
An update on this issue from the Unity side, in case anyone else is still hitting it.

It's been fixed in 2018.2 and up and also 2017.4 (LTS) but won't be fixed for 2018.1 as that release is no longer being supported.

If you cannot migrate to one of these releases, the workaround is basically to remove the "<ExcludeFromResourceIndex>true</ExcludeFromResourceIndex>" tags from the SplashScreen and other "Logo" assets. Either re importing the assets or directly editing the VS project file should work.
